### DISCUSSION TITLE
umi4中如何引入@umijs/preset-ui 插件以及其他umi3的自定义插件

### DISCUSSION THREAD

**Discussion by daisybaicai** [umijs/umi#10285] [Q&A]
umi4中如何引入@umijs/preset-ui 插件以及其他umi3的自定义插件

**Comment by fz6m** [comment]
以前 [plugins](https://github.com/umijs/plugins) 里的插件已经内置到 `@umijs/plugins` 了，可以通过 `plugins: ['@umijs/plugins/dist/*']` 用。

其他插件都需要兼容 umi4 才能用，插件 api 有变化，没兼容的是用不了的。

**Reply by daisybaicai** [reply]
> 以前 [plugins](https://github.com/umijs/plugins) 里的插件已经内置到 `@umijs/plugins` 了，可以通过 `plugins: ['@umijs/plugins/dist/*']` 用。
> 
> 其他插件都需要兼容 umi4 才能用，插件 api 有变化，没兼容的是用不了的。

看了一下umi ui是没有升级 到umi4是吗？
https://github.com/umijs/umi-ui

**Reply by fz6m** [reply]
umi ui 废弃了，不做了。
